samples/osgi-bundles/org.wso2.carbon.uuf.sample.simple-auth.bundle/src/main/java/org/wso2/carbon/uuf/sample/simpleauth/bundle/api/auth/PersistentSessionManager.java

```diff
@@ -0,0 +1,233 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.uuf.sample.simpleauth.bundle.api.auth;
+
+import org.apache.commons.io.FileUtils;
+import org.osgi.service.component.annotations.Component;
+import org.wso2.carbon.uuf.api.auth.Session;
+import org.wso2.carbon.uuf.api.config.Configuration;
+import org.wso2.carbon.uuf.spi.HttpRequest;
+import org.wso2.carbon.uuf.spi.HttpResponse;
+import org.wso2.carbon.uuf.spi.auth.SessionManager;
+import org.wso2.carbon.uuf.spi.auth.User;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import javax.cache.expiry.Duration;
+
+/**
+ * Manages sessions by storing serialized sessions to files.
+ * <p>
+ * This session manager will store the serialized sessions to files. The created session files
+ * could be found in the {UUF-HOME}/tmp/sessions directory
+ * <p>
+ * Please make note to specify the session manager class name in the app.yaml configuration file under the
+ * 'sessionManager' key in order for this session manager to be used in the application.
+ * <p>
+ * eg: sessionManager: "org.wso2.carbon.uuf.sample.simpleauth.bundle.api.auth.PersistentSessionManager"
+ * <p>
+ * The session time-out duration (in seconds) can be specified in the <tt>component.yaml</tt> configuration file under
+ * the
+ * 'sessionTimeoutDuration' key. This will make sure that the session file will be deleted (according to this
+ * implementation) in specified number of seconds.
+ * <p>
+ * eg: sessionTimeoutDuration: 60 # This will keep the session file for 60 seconds
+ *
+ * @since 1.0.0
+ */
+@Component(name = "org.wso2.carbon.uuf.sample.simpleauth.bundle.api.auth.PersistentSessionManager",
+        service = SessionManager.class,
+        immediate = true
+)
+public class PersistentSessionManager implements SessionManager {
+
+    private static final String SESSION_DIR = "tmp/sessions";
```

Comment:
Not {UUF-HOME}/temp directory. We should use [Files.createTempDirectory](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#createTempDirectory-java.lang.String-java.nio.file.attribute.FileAttribute...-) instead.